api/dialogflow_service.py

```diff
@@ -0,0 +1,96 @@
+        response = session_client.detect_intent(session=session, query_input=query_input)
+        return response.query_result
+
+    def retrieve_intent(self, query_result):
+        return Intent(query_result.intent.display_name, query_result.intent_detection_confidence)
+
+    def retrieve_entities(self, nlp_analysis, query_result):
+        for i, entity in query_result.parameters.items():
+            nlp_analysis.add_entity(Entity(entity))
+
+    def add_entity_type(self, name):
+        entity_types_client = dialogflow.EntityTypesClient()
+
+        parent = entity_types_client.project_agent_path(self.project_id)
+        entity_type = dialogflow.types.EntityType(display_name=name, kind=dialogflow.enums.EntityType.Kind.KIND_LIST)
+
+        entity_types_client.create_entity_type(parent, entity_type)
+
+    def add_entity(self, entity_name, entity_value):
+        entity_types_client = dialogflow.EntityTypesClient()
+        entity_type_id = self.get_entity_type_id(entity_name)
+        entity_type_path = entity_types_client.entity_type_path(self.project_id, entity_type_id)
+
+        synonyms = [entity_value]
+        entity = dialogflow.types.EntityType.Entity()
+        entity.value = entity_value
+        entity.synonyms.extend(synonyms)
+
+        entity_types_client.batch_create_entities(entity_type_path, [entity], DialogFlowService.LANGUAGE_CODE)
+
+    def add_entities(self, entity_name, entity_values):
+        entity_types_client = dialogflow.EntityTypesClient()
+        entity_type_id = self.get_entity_type_id(entity_name)
+        entity_type_path = entity_types_client.entity_type_path(self.project_id, entity_type_id)
+
+        entity_list = []
+
+        for value in entity_values:
+            synonyms = [value]
+            entity = dialogflow.types.EntityType.Entity()
+            entity.value = value
+            entity.synonyms.extend(synonyms)
+            entity_list.append(entity)
+
+        entity_types_client.batch_create_entities(entity_type_path, entity_list, DialogFlowService.LANGUAGE_CODE)
+
+    def get_entities(self):
+        entity_types_client = dialogflow.EntityTypesClient()
+        return entity_types_client.list_entity_types(self.parent, DialogFlowService.LANGUAGE_CODE)
+
+    def get_entity_type_id(self, entity_display_name):
+        entity_types_client = dialogflow.EntityTypesClient()
+        entity_types = entity_types_client.list_entity_types(self.parent, DialogFlowService.LANGUAGE_CODE)
+
+        for entity_type in entity_types:
+            if(entity_type.display_name == entity_display_name):
+                return self.get_id_from_path(entity_type.name)
+        raise ValueError('No entity having specified entity_display_name')
+
+    def get_id_from_path(self, path):
+        path_elements = path.split('/')
+        if(len(path_elements) != 5):
+            raise ValueError('Path should be of format: "/projects/projectName/agent/entityTypes/id"')
+        return path_elements[-1]
```

Comment:
General comment for the class, since the class is long, put some comments to specify the method general purpose to distinct if they are used for Dialogflow or the sentence?